KubeArmor/core/kubeUpdate.go

```diff
@@ -509,6 +509,66 @@ func (dm *KubeArmorDaemon) UpdateEndPointWithPod(action string, pod tp.K8sPod) {
 	}
 }
 
+func getTopLevelOwner(obj metav1.ObjectMeta, namespace string, objkind string) (string, string, string, error) {
```

Comment:
@achrefbensaad made changes i think this should cover all cases. let me know what case if left